rca/api_content/content.py

```diff
@@ -274,3 +275,41 @@ def pull_alumni_stories(programme_type_slug=None):
         alumni_stories_blog_page_data + alumni_stories_standard_page_data
     )
     return alumni_stories_data
+
+
+class _BaseContentAPI:
+    def __init__(self, func, cache_key):
+        self.func = func
+        self.cache_key = cache_key
+
+    def fetch_from_api(self):
+        try:
+            data = self.func()
```

Comment:
I've not seen this pattern before but at instantiation wil func = pull_alumni_stories/pull_news_and_events ?

Reply:
Yes, functions are first class objects so they can be assigned/passed around etc, then called when needed.

Reply:
TBH this class inheritance might not be the best pattern, but it separated the code into command and query so they can be used in different contexts.

Passing functions around and calling them dynamically is fine though.